source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -0,0 +1,107 @@
+#include "extensions/common/proxy_protocol/proxy_protocol_header.h"
+
+#include <sstream>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/network/address.h"
+
+#include "common/network/address_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace ProxyProtocol {
+
+void generateV1Header(const std::string& src_addr, const std::string& dst_addr, uint32_t src_port,
+                      uint32_t dst_port, Network::Address::IpVersion ip_version,
+                      Buffer::Instance& out) {
+  std::ostringstream stream;
```

Comment:
Sorry late drive by: my (possibly out of date) understanding is that string streams are pretty slow. We probably would want to move this to an envoy buffer or similar in a later perf pass.